zerver/views/messages.py

```diff
@@ -814,6 +814,7 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
             narrow,
         )
 
+    assert anchor is not None
```

Comment:
So this is to satisfy the succeeding functions using `anchor` as the actual parameter, who don't expect it to be `Optional`.  This was needed in order to mark `anchor` as `Optional` in `get_messages_backend` for the check: `if anchor is None` on line `755`.

Reply:
This `get_messages_backend` parameter change seems appropriate assuming the `default=None` is intended? It would be good to catch this automatically in the `REQ`, but I'm not sure if that's possible. In any case, possibly comment the assert?

Reply:
Added a comment :)